### ISSUE TITLE
OpenjdkAnnotationLocation should allow annotations on single-line type and constructor declarations

### ISSUE THREAD

**Issue by arturict** [checkstyle/checkstyle#21044]
I have read the unreleased check documentation generated for #21030:
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/88196ea_20260729194818/checks/annotation/openjdkannotationlocation.html

`OpenjdkAnnotationLocation` is not available in the latest released CLI yet. The
behavior below is reproduced by the OpenJDK 25 diff report for commit
`4d3dd41d61d75cafdc570d595e24ab4e037eba16`:

https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/4d3dd41d61d75cafdc570d595e24ab4e037eba16_2026003235/reports/diff/openjdk25/index.html

The report was generated with `openjdk_checks.xml`, which configures:

```xml
<module name="OpenjdkAnnotationLocation"/>
```

Three compact declarations in OpenJDK are reported:

```java
@Retention(RUNTIME) @Rat @interface Rat { }
```

Report entry
[#A321](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/4d3dd41d61d75cafdc570d595e24ab4e037eba16_2026003235/reports/diff/openjdk25/index.html#A321):

```text
OpenjdkAnnotationLocation: Annotations must be on a separate line from 'Rat'.
```

```java
@A(b=@B()) class C { }
```

Report entry
[#A319](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/4d3dd41d61d75cafdc570d595e24ab4e037eba16_2026003235/reports/diff/openjdk25/index.html#A319):

```text
OpenjdkAnnotationLocation: Annotations must be on a separate line from 'C'.
```

```java
@TC public <M extends Error, N extends RuntimeException> GenericMethods1(List<@TC E> e) throws @TA M, @TB @TC N { }
```

Report entry
[#A663](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/4d3dd41d61d75cafdc570d595e24ab4e037eba16_2026003235/reports/diff/openjdk25/index.html#A663):

```text
OpenjdkAnnotationLocation: Annotations must be on a separate line from 'GenericMethods1'.
```

This follows the review at
https://github.com/checkstyle/checkstyle/pull/21030#issuecomment-5124975350.

---

**Describe what you expect in detail.**

No `annotation.on.target.line` violation should be reported when annotations and
the complete target declaration are on one physical line for these cases:

* annotation definitions
* class definitions
* constructors

The implementation should preserve violations for multiline declarations. The
three real OpenJDK cases above should become input-file regression tests, and the
fix should be validated with another OpenJDK 25 diff report.

**Comment by arturict** [comment]
Closing as this is covered by #20978 and #21030. These cases are part of the agreed single-line-target behavior, not a separate bug.
